connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/ConnectorHelper.java

```diff
@@ -116,4 +116,52 @@ private static <T> T parseJsonVarsAsTypeOrThrow(
           e);
     }
   }
+
+  /**
+   * Parse a JSON variable as a given type.
+   *
+   * @param jsonVars JSON string coming from {@link
+   *     io.camunda.zeebe.client.api.response.ActivatedJob#getVariables()}
+   * @param variable the name of the variable to parse from the JSON string
+   * @param type the type to parse the variable as
+   * @param <T> the type to parse the variable as
+   * @return an <T> containing the parsed variable, or throw a {@link FeelEngineWrapperException} if
+   *     the variable could not be parsed as the given type
+   */
+  private static <T> T parseJsonVariableAsTypeOrThrow(
+      String jsonVars, String variable, Class<T> type) {
+    try {
+      JsonNode rootNode = OBJECT_MAPPER.readTree(jsonVars);
+      JsonNode variableJsonNode = rootNode.get(variable);
+      return OBJECT_MAPPER.treeToValue(variableJsonNode, type);
+    } catch (Exception e) {
+      throw new FeelEngineWrapperException(
+          String.format(ERROR_CANNOT_PARSE_VARIABLES, jsonVars + '.' + variable, type.getName()),
+          variable,
+          jsonVars,
+          e);
+    }
+  }
+
+  /**
+   * Parse a JSON variable as a given type. Do not throw an exception if the variable could not be
+   * parsed as the given type or if the variable does not exist, but return an empty {@link
+   * Optional}.
+   *
+   * @param jsonVars JSON string coming from {@link
+   *     io.camunda.zeebe.client.api.response.ActivatedJob#getVariables()}
+   * @param variable the name of the variable to parse from the JSON string
+   * @param type the type to parse the variable as
+   * @return an {@link Optional} containing the parsed variable, or an empty {@link Optional} if the
+   *     variable could not be parsed as the given type or if the variable does not exist
+   * @see #parseJsonVariableAsTypeOrThrow(String, String, Class)
+   */
+  public static <T> Optional<T> parseJsonVariableAsType(
+      String jsonVars, String variable, Class<T> type) {
+    try {
+      return Optional.of(parseJsonVariableAsTypeOrThrow(jsonVars, variable, type));
+    } catch (FeelEngineWrapperException e) {
```

Comment:
Maybe worth putting a log statement to be easier identifying why we got an empty result

Reply:
fixed 👍